text/0000-naming-conventions.md

```diff
@@ -0,0 +1,267 @@
+- Start Date: 2018-03-21
+- RFC PR: (leave this empty)
+- WordPress Coding Standards Issue: (leave this empty)
+
+# Summary
+
+This guide is intended to maximize consistency when writing code in the WordPress environment.
+Naming your language constructs is important and should reflect the nature of what it does.
+
+# Basic example
+
+Naming variables, functions, classes, namespaces should be unified across the core, themes and plugins.
```

Comment:
I personally don't agree with that statement. The WPCS is first and foremost a standard to define how WordPress Core development is handled. If plugins & themes opt to follow that same standard, that is perfectly fine.

Reply:
However, plugins & themes are also coded in PHP, and that has its own standard as well.

As long as the two don't perfectly overlap, plugin authors using general-purpose PHP libraries will always be forced to mix&match and choose their own best fit.